app/controllers/api/v1/measurements_controller.rb

```diff
@@ -33,16 +33,30 @@ def index
         end
 
         if params.keys.include? "limit"
-          case params[:limit]
-            when 'first'
-              query = query.where(filter).order(:timestamp)
-            respond_with [query.first]
-            when 'last'
-              query = query.where(filter).order(:timestamp)
-            respond_with [query.last]
+          result = []
+          timelines = []
+          if params[:limit] == 'first'
+            query = query.where(filter).includes(:timeline).order('timestamp ASC')
+          else
+            query = query.where(filter).includes(:timeline).order('timestamp DESC')
           end
+
+          query.each do |m|
+            if timelines.include? m.timeline
```

Comment:
If `if` statement is doint nothing than you can change it into `unless timelines.include? m.timeline` and proceed to line 48.